ext/msgpack/packer.c

```diff
@@ -114,6 +114,40 @@ bool msgpack_packer_try_write_with_ext_type_lookup(msgpack_packer_t* pk, VALUE v
     }
 
     if(ext_flags & MSGPACK_EXT_RECURSIVE) {
+        // HACK: While we call the proc, the current pk->buffer won't be reachable
+        // as it will be stored on the stack.
+        // To ensure all the `mapped_string` reference in that buffer are properly
+        // marked and pined, we copy them all on the stack.
```

Comment:
pined -> pinned
BTW I don't see any pinning here, so AFAIK both the `VALUE` strings and their `char*` can still move.

Reply:
All objects on the stack are pinned in Ruby, so all `VALUE` in the alloca will be pinned.

Whereas if we store them in an array, then the pointers can move (since array supports compaction). Only the array will be pinned in this case.

Reply:
It does not get pinned forever. The pin status of all objects is reset at the start of the GC cycle and gets pinned during marking. Objects on the stack are not treated any differently than other pinned objects.